src/software/jetson_nano/primitive_executor.cpp

```diff
@@ -1,54 +1,38 @@
 #include "software/jetson_nano/primitive_executor.h"
 
+#include "proto/message_translation/tbots_geometry.h"
 #include "proto/primitive.pb.h"
 #include "proto/primitive/primitive_msg_factory.h"
+#include "proto/tbots_software_msgs.pb.h"
+#include "proto/visualization.pb.h"
 #include "software/logger/logger.h"
 #include "software/math/math_functions.h"
 
-
-void PrimitiveExecutor::startPrimitive(const RobotConstants_t& robot_constants,
-                                       const TbotsProto::Primitive& primitive)
+PrimitiveExecutor::PrimitiveExecutor(const double time_step,
+                                     const RobotConstants_t& robot_constants)
+    : current_primitive_(),
+      robot_constants_(robot_constants),
+      hrvo_simulator(static_cast<float>(time_step), robot_constants)
 {
-    robot_constants_   = robot_constants;
-    current_primitive_ = primitive;
 }
 
-Vector PrimitiveExecutor::getTargetLinearVelocity(
-    const TbotsProto::MovePrimitive& move_primitive, const RobotState& robot_state)
+void PrimitiveExecutor::updatePrimitiveSet(
+    const unsigned int robot_id, const TbotsProto::PrimitiveSet& primitive_set_msg)
 {
-    const float LOCAL_EPSILON = 1e-6f;  // Avoid dividing by zero
+    hrvo_simulator.updatePrimitiveSet(primitive_set_msg);
+    // TODO Might be able to improve. don't think can use []
+    current_primitive_ = primitive_set_msg.robot_primitives().at(robot_id);
```

Comment:
Its good to check if the id exists first so this doesn't crash if we get a proto without a primitive for this robot

Reply:
Check added and comment removed